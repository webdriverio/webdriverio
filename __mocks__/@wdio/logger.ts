import { vi } from 'vitest'

export const logMock = {
    error: vi.fn(),
    debug: vi.fn(),
    warn: vi.fn(),
    info: vi.fn(),
    trace: vi.fn(),
    progress: vi.fn()
}
const mock = () => logMock
mock.setLevel = vi.fn()
mock.setLogLevelsConfig = vi.fn()
mock.waitForBuffer = vi.fn()
mock.clearLogger = vi.fn()
export default mock
