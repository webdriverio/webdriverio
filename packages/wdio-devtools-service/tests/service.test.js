import DevToolsService from '../src'

import logger from 'wdio-logger'

jest.mock('../src/commands', () => {
    class CommandHandlerMock {
        constructor () {
            this.cdp = jest.fn()
        }
    }

    return CommandHandlerMock
})

jest.mock('../src/utils', () => {
    let wasCalled = false
    return {
        findChromePort: jest.fn().mockImplementation(() => {
            if (!wasCalled) {
                wasCalled = true
                return 42
            }
            throw new Error('boom')
        }),
        getCDPClient: jest.fn()
    }
})

test('beforeSession', () => {
    const service = new DevToolsService()
    expect(service.isSupported).toBe(false)

    service.beforeSession(null, { browserName: 'firefox' })
    expect(service.isSupported).toBe(false)
    service.beforeSession(null, { browserName: 'chrome', version: 62 })
    expect(service.isSupported).toBe(false)

    service.beforeSession(null, { browserName: 'chrome', version: 65 })
    expect(service.isSupported).toBe(true)
})

test('if not supported by browser', async () => {
    global.browser = { addCommand: jest.fn() }

    const service = new DevToolsService()
    service.isSupported = false

    await service.before()
    expect(global.browser.addCommand.mock.calls).toHaveLength(1)
    expect(global.browser.addCommand.mock.calls[0][0]).toBe('cdp')
    expect(global.browser.addCommand.mock.calls[0][1].toString()).toContain('throw new Error')
})

test('if supported by browser', async () => {


    const service = new DevToolsService()
    service.isSupported = true
    await service.before()
    expect(service.commandHandler.cdp).toBeCalledWith('Network', 'enable')
    expect(service.commandHandler.cdp).toBeCalledWith('Page', 'enable')
})

test('initialization fails', async () => {
    jest.mock('../src/utils', () => ({
        findChromePort: () => {
            throw new Error('boom!')
        },
        getCDPClient: jest.fn()
    }))

    const service = new DevToolsService()
    service.isSupported = true
    await service.before()

    expect(service.commandHandler).toBe(undefined)
    expect(logger().error.mock.calls.pop()[0]).toContain(`Couldn't connect to chrome: Error: boom`)
})
