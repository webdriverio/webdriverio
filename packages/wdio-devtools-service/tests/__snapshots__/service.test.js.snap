// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_emulateDevice 1`] = `
Array [
  Array [
    Object {
      "name": "Nexus 6P",
      "userAgent": "Mozilla/5.0 (Linux; Android 8.0.0; Nexus 6P Build/OPP3.170518.006) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3765.0 Mobile Safari/537.36",
      "viewport": Object {
        "deviceScaleFactor": 3.5,
        "hasTouch": true,
        "height": 732,
        "isLandscape": false,
        "isMobile": true,
        "width": 412,
      },
    },
  ],
]
`;
